src/components/app/UrlManager.js

```diff
@@ -4,26 +4,40 @@
 
 // @flow
 
-import React, { PureComponent } from 'react';
+import * as React from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import explicitConnect from '../../utils/connect';
 import { getIsUrlSetupDone } from '../../reducers/app';
 import { updateUrlState, urlSetupDone, show404 } from '../../actions/app';
+import { urlFromState, stateFromLocation } from '../../url-handling';
 
-type Props = {
-  stateFromLocation: Location => any,
-  urlFromState: any => string,
-  children: any,
-  urlState: any,
-  isUrlSetupDone: boolean,
-  updateUrlState: string => void,
-  urlSetupDone: void => void,
-  show404: string => void,
-};
+import type { ExplicitConnectOptions } from '../../utils/connect';
+import type { UrlState } from '../../types/reducers';
+
+type StateProps = {|
+  +urlState: UrlState,
+  +isUrlSetupDone: boolean,
+|};
+
+type DispatchProps = {|
+  +updateUrlState: typeof updateUrlState,
+  +urlSetupDone: typeof urlSetupDone,
+  +show404: typeof show404,
+|};
+
+type OwnProps = {|
+  +children: React.Node,
+|};
+
+type Props = {|
+  ...OwnProps,
+  ...StateProps,
+  ...DispatchProps,
+|};
```

Comment:
I changed how connected props are constructed to deal with this issue.

```
export type ConnectedProps<
  OwnProps: Object,
  StateProps: Object,
  DispatchProps: Object
> = $ReadOnly<{|
  ...OwnProps,
  ...StateProps,
  ...DispatchProps,
|}>;
```

Which is also nice because we have a consistent interface to deal with Props construction.